sql/src/main/java/org/apache/druid/sql/calcite/planner/DruidPlanner.java

```diff
@@ -261,6 +261,16 @@ private RelNode possiblyWrapRootWithOuterLimitFromContext(
       return root.rel;
     }
 
+    if (root.rel instanceof LogicalSort) {
+      LogicalSort outerSort = (LogicalSort) root.rel;
+      return LogicalSort.create(
+          outerSort.getInput(),
+          outerSort.collation,
+          makeBigIntLiteral(0),
```

Comment:
This won't handle offsets properly, nor will it handle properly the case where the outer limit is higher than the inner limit. Check out `SortCollapseRule` for some logic that handles both of these cases. You might want to create a helper method and use it in both places.

Reply:
pushed some repeated code to extract fetch and offset and fetch combining logic from `SortCollapseRule` to `Calcites`